src/apps/bastion-protocol/helper/bastion-protocol.swap.token-helper.ts

```diff
@@ -0,0 +1,155 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { compact } from 'lodash';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { ETH_ADDR_ALIAS, ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import {
+  buildDollarDisplayItem,
+  buildNumberDisplayItem,
+  buildPercentageDisplayItem
+} from '~app-toolkit/helpers/presentation/display-item.present';
+import { getImagesFromToken } from '~app-toolkit/helpers/presentation/image.present';
+import { ContractType } from '~position/contract.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+import BASTION_PROTOCOL_DEFINITION from '../bastion-protocol.definition';
+import { BastionProtocolComptroller, BastionProtocolContractFactory, BastionProtocolCtoken } from '../contracts';
+
+export type BastionLPTokenDataProps = {
+  swapAddress: string;
+  liquidity: number;
+  fee: number;
+};
+
+type BastionSupplyTokenHelperParams<T = BastionProtocolComptroller, V = BastionProtocolCtoken> = {
+  pools: {
+    swapAddress: string;
+    lpTokenAddress: string;
+    label: string;
+  }[];
+  isCToken?: boolean; // used to determine how to get the price
+  network: Network;
+  appId: string;
+  groupId: string;
+};
+
+@Injectable()
+export class BastionSwapTokenHelper {
```

Comment:
Is it possible to use the `CurvePoolTokenHelper` from `src/apps/curve/helpers`? See `src/apps/saddle/ethereum/saddle.pool.token-fetcher.ts` for an example of a Curve fork that uses those helper classes to avoid code duplication.

Reply:
We don't keep track of the volume at the moment so the value is zero for now. Otherwise, using the `CurvePoolTokenHelper` works fine.